src/components/frontend-engine/frontend-engine.tsx

```diff
@@ -90,8 +91,30 @@ const FrontendEngineInner = forwardRef<IFrontendEngineRef, IFrontendEngineProps>
 	);
 });
 
-export const FrontendEngine = forwardRef<IFrontendEngineRef, IFrontendEngineProps>((props, ref) => (
-	<YupProvider>
-		<FrontendEngineInner {...props} ref={ref} />
-	</YupProvider>
-));
+/**
+ * prevents inferrence
+ * https://stackoverflow.com/questions/56687668/a-way-to-disable-type-argument-inference-in-generics
+ */
+type NoInfer<T, U> = [T][T extends U ? 0 : never];
+/**
+ * The one and only component needed to create your form
+ *
+ * Minimally you will need to set the `data` props which is the JSON schema to define the form
+ */
+export const FrontendEngine = forwardRef<IFrontendEngineRef, IFrontendEngineProps>((props, ref) => {
+	return (
+		<YupProvider>
+			<FrontendEngineInner {...props} ref={ref} />
+		</YupProvider>
+	);
+}) as <V = IYupValidationRule>(
+	props: IFrontendEngineProps<IYupValidationRule | NoInfer<V, IYupValidationRule>> & { ref?: Ref<IFrontendEngineRef> }
```

Comment:
does `V` need to extend `IYupValidationRule`?

Reply:
thanks for raising, after consideration, I decided to change the underlying validation config type to be an intersection of `IYupValidationRule` and a couple of other generics (https://github.com/LifeSG/web-frontend-engine/pull/20/files#diff-74758a5fab71a992c8c83c2ef5dec2dcd28b8422aa7002ba6037a3be86289cdaR103)

the consideration is in most cases, the fields are using `IYupValidationRule` and will occasionally need to add a few more rules hence having the intersection will simplify the typings

with this change, all `V` generic defaults to undefined. 

there is a new `U` generic to allow fields to customise validation rules internally, this prevents getting overwritten by the custom validation type.